packages/core/src/queries/log.ts

```diff
@@ -10,49 +11,87 @@ import { buildInsertIntoWithPool } from '#src/database/insert-into.js';
 
 const { table, fields } = convertToIdentifiers(Logs);
 
-type LogCondition = {
+type AuditLogCondition = {
   logKey?: string;
   applicationId?: string;
   userId?: string;
+};
+
+type WebhookLogCondition = {
+  logKey?: string;
   hookId?: string;
   startTimeExclusive?: number;
 };
 
-const buildLogConditionSql = (logCondition: LogCondition) =>
-  conditionalSql(logCondition, ({ logKey, applicationId, userId, hookId, startTimeExclusive }) => {
-    const subConditions = [
-      conditionalSql(logKey, (logKey) => sql`${fields.key}=${logKey}`),
+const buildAuditLogConditionSql = (logCondition: AuditLogCondition) =>
```

Comment:
Is it possible to use function overload to avoid defining multiple query methods?